Sources/JavaKit/Optional+JavaOptional.swift

```diff
@@ -0,0 +1,81 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift.org project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Swift.org project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+extension Optional where Wrapped: AnyJavaObject {
+  func toJavaOptional() -> JavaOptional<Wrapped> {
+    return try! JavaClass<JavaOptional<Wrapped>>().ofNullable(self?.as(JavaObject.self)).as(JavaOptional<Wrapped>.self)!
+  }
+
+  init(javaOptional: JavaOptional<Wrapped>?) {
+    if let javaOptional {
+      self = javaOptional.isPresent() ? javaOptional.get().as(Wrapped.self) : Optional<Wrapped>.none
+    } else {
+      self = nil
+    }
+  }
+}
+
+extension Optional where Wrapped == Double {
+  func toJavaOptional() -> JavaOptionalDouble {
+    if let self {
+      return try! JavaClass<JavaOptionalDouble>().of(self)!
+    } else {
+      return try! JavaClass<JavaOptionalDouble>().empty()!
+    }
+  }
+
+  init(javaOptional: JavaOptionalDouble?) {
+    if let javaOptional {
+      self = javaOptional.isPresent() ? javaOptional.getAsDouble() : nil
+    } else {
+      self = nil
+    }
+  }
+}
+
+extension Optional where Wrapped == Int32 {
+  func toJavaOptional() -> JavaOptionalInt {
+    if let self {
+      return try! JavaClass<JavaOptionalInt>().of(self)!
+    } else {
+      return try! JavaClass<JavaOptionalInt>().empty()!
+    }
+  }
+
+  init(javaOptional: JavaOptionalInt?) {
+    if let javaOptional {
+      self = javaOptional.isPresent() ? javaOptional.getAsInt() : nil
+    } else {
+      self = nil
+    }
+  }
+}
+
+extension Optional where Wrapped == Int64 {
```

Comment:
Should we also produce those for `UInt` versions? Java doesn't have unsigned but may be convenient for Swift hm

Reply:
Would these map to JavaOptional<UInt32> or did you have something else in mind? There is no built-in type, but it could be nice (we can map the boxed types if we don't already)

Reply:
I don't see any unsigned long wrapper types (could have missed them in the docs), so I worry about mapping these as we would be changing the signedness of them

Reply:
Ok let's let those be :)